src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -105,9 +145,32 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     return true;
   }
 
+  /**
+   * vectorize replacementDocument as needed, only when document is found will be used by
+   * findOneAndReplace
+   *
+   * @param dataVectorizer
+   */
+  public void vectorizeTheReplacementDocument(DataVectorizer dataVectorizer) {
+    try {
+      dataVectorizer
```

Comment:
Same code as above method can be merge the methods?

Reply:
method body is different though, leave as it since it is more clearer